CMakeLists.txt

```diff
@@ -462,8 +462,8 @@ endif()
 
 # require minimally the bundled version
 find_package(Boost 1.61 COMPONENTS thread system regex random program_options date_time iostreams REQUIRED)
-include_directories(${Boost_INCLUDE_DIRS})
-include_directories(${PROJECT_BINARY_DIR}/include)
+include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
```

Comment:
It's not required with Clang, but I'd want it with GCC anyway. Getting warnings from the Boost include files is just noise for us and distracts from keeping our own code clean.

(We ought not get used to paged of warnings scrolling by.)